src/firefly/js/ui/CompleteButton.jsx

```diff
@@ -33,59 +33,52 @@ function addComponentData(inR={}, registeredComponents={}) {
 }
 
 
-function validUpdate(valid,onSuccess,onFail,registeredComponents, closeOnValid,groupKey,
-                     dialogId, groupsToUse, includeUnmounted= false) {
-    const funcToCall = valid ? onSuccess : onFail;
-    let continueValid= true;
-
-    if (Array.isArray(groupKey)) {
-        const groupsToValidate= groupsToUse();
-        const requestObj = groupsToValidate.reduce( (obj,groupKey) => {
-            obj[groupKey]= getFieldGroupResults(groupKey,includeUnmounted);
-            return obj;
-        },{});
-        if (funcToCall) funcToCall(addComponentData(requestObj,registeredComponents));
-    }
-    else {
-        const request = getFieldGroupResults(groupKey,includeUnmounted);
-        if (funcToCall) continueValid= funcToCall(addComponentData(request,registeredComponents));
-    }
+async function collectInputs(registeredComponents, groupKey, includeUnmounted= false) {
 
-    const stillValid= valid && (continueValid ?? true);
+    if (!groupKey) return [{}, true];   // if no groupKey, then no parameters to collect and therefore pass validation
 
-    if (stillValid && dialogId && closeOnValid) dispatchHideDialog(dialogId);
-}
+    const groups = isFunction(groupKey) ? groupKey() : groupKey;
+    const valid = await validateFieldGroup(groups, includeUnmounted);
```

Comment:
notice groups can be an array or a string `validateFieldGroup` accepts both. groupKey can be an array.

Reply:
Yes.  That's why I resolved `groupKey` in case it's a func.